docs/DeveloperGuide.adoc

```diff
@@ -277,106 +277,365 @@ Refer to the guide <<DevOps#, here>>.
+
+> Use case ends
+
+***Extensions***
+
+2a. The list is empty
+
+Use case ends
+
+3a. The **Name** provided is invalid
+
+3a1. TravelPal shows an error message
+
+3a2. TravelPal does not delete any **Day**
+
+Use case ends
+
+**User case: UC7 – Add Event**
+
+***MSS***
+
+1.  User chooses to add a **Event** to a specified **Day**
+
+2.  User _edits the event (UC5)_
+
+3.  TravelPal saves the **Event**
+
+***Extensions***
+
+3a **Event** added clashes with other **Events** in the **Day**
+
+3a1. TravelPal shows an error message
+
+3a2. TravelPal does not discard information the user has provided
+
+3a3. TravelPal displays the **Edit Event** page containing the user’s
+input
+
+3a4. TravelPal requests the user to change the date of the **Event**
+
+Steps 3a1 – 3a4 are repeated until the user provided non clashing date
+
+**User case UC8 – Edit Event**
+
+***MSS***
+
+1.  User requests to edit specific **Day**
+
+2.  TravelPal shows the **Edit Day** page with fields to enter
 
-_{More to be added}_
+3.  User edits information in the specified **Day**
+
+4.  User submits and confirms the edit
+
+> Use case ends
+
+***Extensions***
+
+3a. User enters an invalid field
+
+3a1. TravelPal shows an error message
+
+3a2. TravelPal does not edit invalid field
+
+Use case continues at step 2
+
+3b. User requests to list of **Events** in the **trip**
+
+3b1. TravelPal shows a list of **Events** to the user (can be empty)
+
+3b2. User chooses to _add/edit/delete (UC 7/8/9) **Event**_
+Use case continues at step 4
+
+4b. User leaves necessary information empty
+
+4a1. TravelPal shows an error message
+
+4a2. TravelPal does not submit the details and does not confirm the edit
+
+4a3. User enters new data
+
+Steps 4a1-4a3 are repeated until the data entered are non empty
+
+Use case ends.
+
+**User case UC9 – Delete Event**
+
+***MSS***
+
+1.  User requests to delete a specific **Event** in the list
+
+2.  TravelPal deletes the **Event**
+
+> Use case ends
+
+***Extensions***
+
+2a. The list is empty
+
+Use case ends
+
+3a. The **Name** provided is invalid
+
+3a1. TravelPal shows an error message
+
+3a2. TravelPal does not delete any **Event**
+
+Use case ends
 
 [appendix]
 == Non Functional Requirements
 
-.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1.  Should work on any [mainstream OS] as
+    long as it has Java 11 or above installed.
 
-_{More to be added}_
+2.  A user with above average typing speed for regular English text
+    (i.e. not code, not system admin commands) should be able to
+    accomplish most of the tasks faster using commands than using the
+    mouse.
```

Comment:
May want to make this a bit more concrete so that it's easier for developers to test!